lib/processors/bundlers/flexChangesBundler.js

```diff
@@ -10,19 +10,77 @@ const resourceFactory = require("@ui5/fs").resourceFactory;
  * @param {module:@ui5/fs.Resource[]} parameters.resources List of resources to be processed
  * @param {Object} parameters.options Options
  * @param {string} parameters.options.pathPrefix Prefix for bundle path
+ * @param {string} parameters.options.hasFlexBundleVersion true if minUI5Version >= 1.73 than create flexibility-bundle.json
  * @returns {Promise<module:@ui5/fs.Resource[]>} Promise resolving with flex changes bundle resources
  */
 module.exports = function({resources, options}) {
+	let bundleName = "changes-bundle.json";
+
 	function sortByTimeStamp(a, b) {
 		return a.creation > b.creation ? 1 : -1;
 	}
 
-	function sortAndStringify(changesContent) {
+	/**
+	 * bundle changes resource to json string
+	 *
+	 * @param {Array} changesContent Array of resources files
+	 * @returns {string} Json sting of changes and control variants
+	 */
+	function sortAndStringifyInFlexFormat(changesContent) {
 		changesContent = changesContent.sort(sortByTimeStamp);
-		changesContent = changesContent.map(function(change) {
-			return JSON.stringify(change);
+		const changes = [];
+		const variantDependentControlChanges = [];
+		const compVariants = [];
+		const variants = [];
+		const variantChanges = [];
+		const variantManagementChanges = [];
+
+		changesContent.forEach(function(content) {
+			switch (content.fileType) {
+			case "change":
+				if (content.appDescriptorChange && (content.appDescriptorChange === "true" || content.appDescriptorChange == true)) {
+					break;
+				}
+				if (content.variantReference && content.variantReference !== "") {
+					variantDependentControlChanges.push(content);
+				} else {
+					changes.push(content);
+				}
+				break;
+			case "variant":
+				compVariants.push(content);
+				break;
+			case "ctrl_variant":
+				variants.push(content);
+				break;
+			case "ctrl_variant_change":
+				variantChanges.push(content);
+				break;
+			case "ctrl_variant_management_change":
+				variantManagementChanges.push(content);
+				break;
+			}
 		});
-		return changesContent;
+
+		if (!options.hasFlexBundleVersion && (compVariants.length != 0 || variants.length != 0 || variantChanges.length != 0 || variantDependentControlChanges.length != 0 || variantManagementChanges.length != 0)) {
+			throw new Error("There is some control variant changes in the changes folder. This only works with a minUI5Version 1.73.0. Please update the minUI5Version in the manifest.json to 1.73.0 or higher");
```

Comment:
Either "There is some control variant change..." (all singular) or "There are some control variant changes...." (all plural)

Reply:
done